cgo: github.com/andlabs/ui - #include <gtk/gtk.h> not found

this is on linux/amd64

```
[cryptix@higgs ~/devel/projx/vendor/src/github.com/andlabs/ui] git log -1
commit e34c561ed5bedeb180437ec165882b98d70d38c1
Author: Pietro Gagliardi <pietro10@mac.com>
Date:   Thu Apr 16 20:33:28 2015 -0400

    Split the rewrite into a new repository.
[cryptix@higgs ~/devel/projx/vendor/src/github.com/andlabs/ui] go build -v
_/home/cryptix/devel/projx/vendor/src/github.com/andlabs/ui
[cryptix@higgs ~/devel/projx/vendor/src/github.com/andlabs/ui] gb build .
In file included from /home/cryptix/devel/projx/vendor/src/github.com/andlabs/ui/area_unix.go:13:0:
gtk_unix.h:23:21: fatal error: gtk/gtk.h: No such file or directory
 #include <gtk/gtk.h>
                     ^
compilation terminated.
FATAL command "build" failed: # /home/cryptix/gotip/pkg/tool/linux_amd64/cgo -objdir /tmp/gb298671284/github.com/andlabs -- -I /home/cryptix/devel/projx/vendor/src/github.com/andlabs/ui area_unix.go button_unix.go checkbox_unix.go common_unix.go container_unix.go control_unix.go dialog_unix.go group_unix.go image_unix.go label_unix.go progressbar_unix.go spinbox_unix.go tab_unix.go table_unix.go textbox_unix.go textfield_unix.go uitask_unix.go window_unix.go: exit status 2
```

I tried it again with #201, still getting the `no such file..`-error.

```
# using the basicwindow example with andlabs/ui vendored
[cryptix@higgs ~/play/devel/andlabsEx/src/andlabsEx] gb build -v
DEBUG project root "/home/cryptix/play/devel/andlabsEx"
DEBUG matchPackages: [/home/cryptix/play/devel/andlabsEx/src] ...
DEBUG args: [andlabsEx]
DEBUG loadPackage: github.com/andlabs/ui true (/home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui)
DEBUG loadPackage: andlabsEx true (/home/cryptix/play/devel/andlabsEx/src/andlabsEx)
DEBUG buildPackage: andlabsEx
DEBUG buildPackage: github.com/andlabs/ui
DEBUG cd /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui; [/home/cryptix/gotip/pkg/tool/linux_amd64/cgo -objdir /tmp/gb727636050/github.com/andlabs -- -I /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui area_unix.go button_unix.go checkbox_unix.go common_unix.go container_unix.go control_unix.go dialog_unix.go group_unix.go image_unix.go label_unix.go progressbar_unix.go spinbox_unix.go tab_unix.go table_unix.go textbox_unix.go textfield_unix.go uitask_unix.go window_unix.go]
In file included from /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui/area_unix.go:13:0:
gtk_unix.h:23:21: fatal error: gtk/gtk.h: No such file or directory
compilation terminated.
DEBUG {main andlabsEx /home/cryptix/play/devel/andlabsEx/src/andlabsEx} is a main package, not installing
DEBUG removing work directory: /tmp/gb727636050
DEBUG build duration: 18.082837ms map[]
FATAL command "build" failed: # /home/cryptix/gotip/pkg/tool/linux_amd64/cgo -objdir /tmp/gb727636050/github.com/andlabs -- -I /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui area_unix.go button_unix.go checkbox_unix.go common_unix.go container_unix.go control_unix.go dialog_unix.go group_unix.go image_unix.go label_unix.go progressbar_unix.go spinbox_unix.go tab_unix.go table_unix.go textbox_unix.go textfield_unix.go uitask_unix.go window_unix.go: exit status 2
```

Thanks. I hope to have some time to work on this later this week.

On Mon, Jun 15, 2015 at 2:43 PM, Henry <notifications@github.com> wrote:

> I tried it again with #201 <https://github.com/constabulary/gb/pull/201>,
> still getting the no such file..-error.
>
> # using the basicwindow example with andlabs/ui vendored
> [cryptix@higgs ~/play/devel/andlabsEx/src/andlabsEx] gb build -v
> DEBUG project root "/home/cryptix/play/devel/andlabsEx"
> DEBUG matchPackages: [/home/cryptix/play/devel/andlabsEx/src] ...
> DEBUG args: [andlabsEx]
> DEBUG loadPackage: github.com/andlabs/ui true (/home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui)
> DEBUG loadPackage: andlabsEx true (/home/cryptix/play/devel/andlabsEx/src/andlabsEx)
> DEBUG buildPackage: andlabsEx
> DEBUG buildPackage: github.com/andlabs/ui
> DEBUG cd /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui; [/home/cryptix/gotip/pkg/tool/linux_amd64/cgo -objdir /tmp/gb727636050/github.com/andlabs -- -I /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui area_unix.go button_unix.go checkbox_unix.go common_unix.go container_unix.go control_unix.go dialog_unix.go group_unix.go image_unix.go label_unix.go progressbar_unix.go spinbox_unix.go tab_unix.go table_unix.go textbox_unix.go textfield_unix.go uitask_unix.go window_unix.go]
> In file included from /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui/area_unix.go:13:0:
> gtk_unix.h:23:21: fatal error: gtk/gtk.h: No such file or directory
> compilation terminated.
> DEBUG {main andlabsEx /home/cryptix/play/devel/andlabsEx/src/andlabsEx} is a main package, not installing
> DEBUG removing work directory: /tmp/gb727636050
> DEBUG build duration: 18.082837ms map[]
> FATAL command "build" failed: # /home/cryptix/gotip/pkg/tool/linux_amd64/cgo -objdir /tmp/gb727636050/github.com/andlabs -- -I /home/cryptix/play/devel/andlabsEx/vendor/src/github.com/andlabs/ui area_unix.go button_unix.go checkbox_unix.go common_unix.go container_unix.go control_unix.go dialog_unix.go group_unix.go image_unix.go label_unix.go progressbar_unix.go spinbox_unix.go tab_unix.go table_unix.go textbox_unix.go textfield_unix.go uitask_unix.go window_unix.go: exit status 2
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/issues/114#issuecomment-111920931>.
>

--- a/cgo.go
+++ b/cgo.go
@@ -18,15 +18,13 @@ func (t cgoTarget) Objfile() string { return string(t) }
 func (t cgoTarget) Result() error   { return nil }
 
 // rungcc1 invokes gcc to compile cfile into ofile
-func rungcc1(pkg *Package, ofile, cfile string) Target {
-	_, cgoCFLAGS, _, _ := cflags(pkg, true)
-	args := []string{"-fPIC", "-m64", "-pthread", "-fmessage-length=0",
+func rungcc1(pkg *Package, cgoCFLAGS []string, ofile, cfile string) Target {
+	args := []string{"-g", "-O2", "-fPIC", "-m64", "-pthread", "-fmessage-length=0",
 		"-I", pkg.Dir,
 		"-I", filepath.Dir(ofile),
 	}
 	args = append(args, cgoCFLAGS...)
 	args = append(args,
-		"-g", "-O2",
 		"-o", ofile,
 		"-c", cfile,
 	)

--- a/cgo14.go
+++ b/cgo14.go
@@ -16,7 +16,7 @@ func cgo(pkg *Package) ([]ObjTarget, []string) {
 	fn := func(t ...ObjTarget) ([]ObjTarget, []string) {
 		return t, nil
 	}
-	_, _, cgoCFLAGS, cgoLDFLAGS := cflags(pkg, true)
+	_, cgoCFLAGS, _, cgoLDFLAGS := cflags(pkg, false)
 	pcCFLAGS, pcLDFLAGS, err := pkgconfig(pkg)
 	if err != nil {
 		return fn(ErrTarget{err})
@@ -51,7 +51,7 @@ func cgo(pkg *Package) ([]ObjTarget, []string) {
 	for _, cfile := range cfiles {
 		ofile := filepath.Join(pkg.Objdir(), stripext(filepath.Base(cfile))+".o")
 		ofiles = append(ofiles, ofile)
-		if err := rungcc1(pkg, ofile, cfile).Result(); err != nil {
+		if err := rungcc1(pkg, cgoCFLAGS, ofile, cfile).Result(); err != nil {
 			return fn(ErrTarget{err})
 		}
 	}
@@ -91,6 +91,8 @@ func runcgo1(pkg *Package, cflags, ldflags []string) error {
 		"--",
 		"-I", pkg.Dir,
 	}
+	args = append(args, cflags...)
+	args = append(args, ldflags...)
 	args = append(args, pkg.CgoFiles...)
 
 	cgoenv := []string{

--- a/cgo15.go
+++ b/cgo15.go
@@ -16,7 +16,7 @@ func cgo(pkg *Package) ([]ObjTarget, []string) {
 	fn := func(t ...ObjTarget) ([]ObjTarget, []string) {
 		return t, nil
 	}
-	_, _, cgoCFLAGS, cgoLDFLAGS := cflags(pkg, true)
+	_, cgoCFLAGS, _, cgoLDFLAGS := cflags(pkg, false)
 	pcCFLAGS, pcLDFLAGS, err := pkgconfig(pkg)
 	if err != nil {
 		return fn(ErrTarget{err})
@@ -46,7 +46,7 @@ func cgo(pkg *Package) ([]ObjTarget, []string) {
 	for _, cfile := range cfiles {
 		ofile := filepath.Join(pkg.Objdir(), stripext(filepath.Base(cfile))+".o")
 		ofiles = append(ofiles, ofile)
-		targets = append(targets, rungcc1(pkg, ofile, cfile))
+		targets = append(targets, rungcc1(pkg, cgoCFLAGS, ofile, cfile))
 	}
 
 	for _, t := range targets {
@@ -89,9 +89,10 @@ func runcgo1(pkg *Package, cflags, ldflags []string) error {
 		"-I", objdir,
 		"-I", pkg.Dir,
 	}
+	args = append(args, cflags...)
+	args = append(args, ldflags...)
 	args = append(args, pkg.CgoFiles...)
 
-	// Update $CGO_LDFLAGS with p.CgoLDFLAGS.
 	cgoenv := []string{
 		"CGO_CFLAGS=" + strings.Join(quoteFlags(cflags), " "),
 		"CGO_LDFLAGS=" + strings.Join(quoteFlags(ldflags), " "),

--- a/wercker.yml
+++ b/wercker.yml
@@ -32,13 +32,14 @@ build:
           gb vendor -R integration-tests/sqlite3 restore
           gb build -R integration-tests/sqlite3
 
-    - script:
-        name: integration test/go-sdl2
-        code: |
-          cd integration-tests/go-sdl2
-          bash -x setup.bash
-          gb vendor -v restore
-          gb build
+# disabled, no access to libsdl2-dev on wercker
+#    - script:
+#        name: integration test/go-sdl2
+#        code: |
+#          cd integration-tests/go-sdl2
+#          bash -x setup.bash
+#          gb vendor -v restore
+#          gb build
 
     - script:
         name: intergration test/goczmq

--- a/wercker.yml
+++ b/wercker.yml
@@ -33,6 +33,14 @@ build:
           gb build -R integration-tests/sqlite3
 
     - script:
+        name: integration test/go-sdl2
+        code: |
+          cd integration-tests/go-sdl2
+          bash -x setup.bash
+          gb vendor -v restore
+          gb build
+
+    - script:
         name: intergration test/goczmq
         code: |
           cd integration-tests/goczmq

